### ISSUE TITLE
List of prefix_inspections duplicated

### ISSUE THREAD

**Issue by ChristianTackeGSI** [LLNL/spack#17457]
spack currently has two places that list `prefix_inspections`:

1. `etc/spack/defaults/modules.yaml` (and `etc/spack/defaults/darwin/modules.yaml`) contains `modules:prefix_inspections`.
2. `lib/spack/spack/user_environment.py:prefix_inspections()` has nearly the same list, just as python code.

There should be only one place to maintain such a list.

**Comment by adamjstewart** [comment]
I completely agree with this. I believe the original idea was that we wanted Spack to work even if a user deletes the default configuration files. But code duplication is a slippery slope.

**Comment by ChristianTackeGSI** [comment]
IMHO `spack load` is its own module system – the "Spack Module System" – with `spack load`, `spack unload`, and `spack find --loaded` being the major user interaction points.
In that sense, it should also obey the config in `modules.yaml`, including `prefix_inspections`.
